### ISSUE TITLE
local / declare, etc. can't be aliased

### ISSUE THREAD

**Issue by andychu** [oilshell/oil#392]
From #236 by @lhoursquentin

I'm not sure what I think about this yet, but it seems to be a separate issue.

> For instance mksh has by default an alias to replace local by typeset

```
$ mksh -c alias | grep local
local='\\builtin typeset'
```

**Comment by andychu** [comment]
Also here is my snippet from  Zulip showing why `declare` has to be statically parsed sometimes.

I think what other shells do is set some flags in words for "nosplit" or something like that.


```
$ d=declare
$ x='a b'

$ declare x=$x  # statically parsed, shell knows it's declare, no word splitting
$ echo $x
a b

$ $d x=$x  # dynamically parsed, word splitting
$ echo $x
a
```

**Comment by andychu** [comment]
Fixed with 0.7.pre2
